src/wallet.js

```diff
@@ -1586,6 +1601,127 @@ const wallet = {
   getTokenIndex(token_data) {
     return token_data & TOKEN_INDEX_MASK;
   },
+
+  /**
+   * Save rewardSpendMinBlocks variable from server
+   *
+   * @param {number} rewardSpendMinBlocks
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  updateRewardLockConstant(rewardSpendMinBlocks) {
+    this._rewardSpendMinBlocks = rewardSpendMinBlocks;
+  },
+
+  /**
+   * Return the minimum blocks required to unlock reward
+   *
+   * @return {number} Minimum blocks required to unlock reward
+   *
+   * @throws {ConstantNotSet} If the weight constants are not set yet
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getRewardLockConstant() {
+    if (this._rewardSpendMinBlocks === null) {
+      throw new ConstantNotSet('Reward block minimum blocks constant not set');
+    }
+    return this._rewardSpendMinBlocks;
+  },
+
+  /**
+   * Clear rewardSpendMinBlocks constants
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  clearRewardLockConstant() {
+    this._rewardSpendMinBlocks = null;
+  },
+
+  /**
+   * Method called when websocket connection is opened
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  websocketOpened() {
+    this.subscribeAllAddresses();
+    this.addMetricsListener();
+  },
+
+  /**
+   * Method called when websocket connection is closed
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  websocketClosed() {
+    this.removeMetricsListener();
+  },
+
+  /**
+   * Start listening dashboard ws messages from full node
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  addMetricsListener() {
+    WebSocketHandler.on('dashboard', this.handleWebsocketDashboard);
+  },
+
+  /**
+   * Stop listening dashboard ws messages from full node
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  removeMetricsListener() {
+    WebSocketHandler.removeListener('dashboard', this.handleWebsocketDashboard);
+  },
+
+  /**
+   * Method called when received dashboard ws message from full node
+   * Right now we just update the network height
+   *
+   * @param {Object} data Metrics object ws data with 'height' key
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  handleWebsocketDashboard(data) {
+    // So far we just use the height of the network and update in the variable
+    wallet.updateNetworkHeight(data.height);
+  },
+
+  /**
+   * Update network height variable
+   *
+   * @param {number} networkHeight
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  updateNetworkHeight(networkHeight) {
+    if (networkHeight !== this._networkHeight) {
+      this._networkHeight = networkHeight;
+      WebSocketHandler.emit('height_updated', networkHeight);
+    }
+  },
+
+  /**
+   * Return the network height
+   *
+   * @return {number} Network height
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getNetworkHeight() {
+    return this._networkHeight;
```

Comment:
Should throw `ConstantNotSet` if not set.

Reply:
It's not exactly a constant, that's why I am handling it differently. I don't want a bad ws connection to make the app crashes when calculating balance, that's why I didn't raise the exception.

I feel it's not necessary and wouldn't be bad for the wallet to calculate the balance considering networkHeight 0 instead of throwing an exception.

Reply:
If we make the API call when we start the wallet, it'll never crash. It's kind of the same treatment we use for the variables we get from version API. We require them to be set.

Reply:
It's easy to add this API call but I still think it's a waste. This variable is totally different from the constants. If the ws is not working the wallet won't work because it will show as 'Offline' and if it's working we will be receiving and updating the height value.

Reply:
Do we get the height immediately after opening the ws or only when there's a new block? If it's the latter, it could take a while before getting the height. Average time is 30s, but it's not uncommon to take more than one minute

Reply:
The metrics ws sends data every 5 seconds, so that's the maximum time we would be outdated. I say it depends on the implementation because in the wallet desktop we start the ws connection before starting the wallet (this might not be good be when we start the wallet we always have the ws data updated). In the mobile wallet we start the ws connection when loading txs, so it's basically the time to load the txs from the server is the time to receive the first ws message.

Reply:
Ah ok, then I think it's fine.